core/src/smartcontracts/wasm.rs

```diff
@@ -327,258 +353,211 @@ pub mod state {
             .build()
     }
 
-    /// Common data for states
-    pub struct Common<'wrld> {
+    /// State for most common operations.
+    /// Generic over concrete state kind and containing common data.
+    pub struct CommonState<K> {
         pub(super) authority: AccountId,
         pub(super) store_limits: StoreLimits,
-        pub(super) wsv: &'wrld mut WorldStateView,
         /// Span inside of which all logs are recorded for this smart contract
         pub(super) log_span: Span,
+        pub(super) executed_queries: HashSet<QueryId>,
+        /// Concrete kind of state
+        pub(super) kind: K,
```

Comment:
Why do you choose to use `CommonState<Kind<Subkind>>` instead of `CommonState<WsvType, ExecutableType>`?

Reply:
Interesting, I'll try

Reply:
Done